trinity/extensibility/plugin.py

```diff
@@ -85,35 +90,41 @@ def name(self) -> str:
     def logger(self) -> logging.Logger:
         return logging.getLogger('trinity.extensibility.plugin.BasePlugin#{0}'.format(self.name))
 
+    @property
+    def event_bus(self) -> Endpoint:
+        if self.context is None:
+            raise EventBusNotReady("Tried accessing ``event_bus`` before ``ready`` was called")
+
+        return self.context.event_bus
+
     def set_context(self, context: PluginContext) -> None:
         """
         Set the :class:`~trinity.extensibility.plugin.PluginContext` for this plugin.
         """
         self.context = context
 
-    def configure_parser(self, arg_parser: ArgumentParser, subparser: _SubParsersAction) -> None:
+    def ready(self) -> None:
         """
-        Called at startup, giving the plugin a chance to amend the Trinity CLI argument parser
+        Called after the plugin received its context and is ready to bootstrap itself.
         """
         pass
 
-    def handle_event(self, activation_event: BaseEvent) -> None:
+    def configure_parser(self, arg_parser: ArgumentParser, subparser: _SubParsersAction) -> None:
         """
-        Notify the plugin about an event, giving it the chance to do internal accounting right
-        before :meth:`~trinity.extensibility.plugin.BasePlugin.should_start` is called
+        Called at startup, giving the plugin a chance to amend the Trinity CLI argument parser
         """
-
         pass
 
-    def should_start(self) -> bool:
+    def handle_start(self) -> None:
         """
-        Return ``True`` if the plugin should start, otherwise return ``False``
+        Prepare the plugin to get started and eventually cause ``start`` to get called.
         """
-
-        return False
-
-    def _start(self) -> None:
+        self.running = True
         self.start()
+        self.event_bus.broadcast(
+            PluginStartedEvent(type(self))
+        )
+        self.logger.info("Plugin started: %s", self.name)
 
     def start(self) -> None:
```

Comment:
I'm ok with `run` but I think I like `start` more since `stop` is a natural opposite.

Reply:
Right, then let's leave this be called `start` and rename `handle_start` to `boot`